### PR TITLE
Changes from Pronet data aggregation server

### PR DESCRIPTION
**Pull request by kcho**
Summary of changes:

1. `lochness.xnat` module to save `.zip` files rather than the raw dicom directory. Saving a large number of dicom files under PHOENIX significantly delayed `aws sync` step. AMPSCZ-yaxil was updated along with this change. https://github.com/AMP-SCZ/yaxil/commit/094fd3464e8de2922cfe902b3cc7452a24e19b19

2. `lochness.redcap` now has the functions to be used in the future for the repeated instrument measure. These functions are not called in the current `sync.py` pipeline, but we will use these functions once DPACC decides to move on with the repeated instrument measure.

3. `lochness.transfer` module updated to only sync the data files selected by the cmdline options. eg) if `box` is selected as the data source when executing the `sync.py`, only the datatypes included in the `box` will get synced. This update is required for faster testing and parallelization in the future version.

4. `lochness.redcap` also saves the redcap metadata as a csv file.

### CODE REVIEW

**Comment by kcho** [comment]
@tashrifbillah This is the working version of lochness at the Pronet aggregation server. It includes all changes not added to the `master` branch. Your quick review would be very helpful. Thanks.

**Comment by kcho** [comment]
This PR will be merged to `master` by the end of next week.

File: lochness/cleaner/__init__.py [hunk 1]
```diff
@@ -24,12 +40,12 @@ def get_ok2remove_df_from_s3_log(phoenix_root: Path,
...
     # since lochness.xnat pulls a zip file to a temporary file and extracts
-    # to a directory, a root directory of the saved MRI data needs to be added 
-    mri_root_names = df_okay_to_remove[
-            df_okay_to_remove['datatypes']=='mri'][
-                    'source'].str.extract(
-                            r'([\w/]*/mri/\w+)/').drop_duplicates()
+    # to a directory, a root directory of the saved MRI data needs to be added
+    mri_df = df_okay_to_remove[df_okay_to_remove['datatypes'] == 'mri']
+    mri_root_names = mri_df['source'].str.extract(r'([\w/]*/mri/\w+)/')
+    mri_root_names.drop_duplicates(inplace=True)
```
Comment (tashrifbillah) [inline lochness/cleaner/__init__.py:48]: Did you break it down to ease debugging?

File: lochness/utils/source_check.py [hunk 2]
```diff
@@ -114,6 +123,8 @@ def check_list_all_redcap_subjects(project_name: str,
 
     df = pd.DataFrame(content_dict_list)
     if len(df) > 1:
+        df.columns = ['subject', '_', 'consent_date']
```
Comment (tashrifbillah) [inline lochness/utils/source_check.py:126]: I don't see the use of these three columns in `get_xnat_credentials_from_config()` at all. Can you comment?

**Review by tashrifbillah — APPROVED** [review]
I trust Kevin to have adequately tested this PR.

File: lochness/cleaner/__init__.py (hunk 1 above)
Comment (kcho) [reply lochness/cleaner/__init__.py:48]: Yes, exactly.

File: lochness/utils/source_check.py (hunk 2 above)
Comment (kcho) [reply lochness/utils/source_check.py:126]: The `source_check.py` screens each data sources like XNAT, REDCap, and Box etc. These columns you highlighted extracts subject and corresponding consent date from the REDCap, which is not directly related with the `get_xnat_credentials_from_config`.

**Pull request merged**